modules/docs/arrow-docs/docs/docs/typeclasses/traverse/README.md

```diff
@@ -9,8 +9,259 @@ permalink: /docs/typeclasses/traverse/
+}
+
+interface Credential
+data class Profile(val id: String)
+data class User(val id: String, val name: String)
+```
+
+`parseInt`: is a function that will try to convert an `String` parameter `s` to an `Int`, it if succeeds it will return the number inside the `Some`. If it fails, it will return `None`.
+
+ `validateLogin`: This function can be something that could either fail or be a sauccessful operation when validating login credentials */
+
+`userInfo`: This function could be something that could return a Profile asynchronously.
+
+```kotlin:ank:silent
+interface OurSideEffectsFunctions {
+    fun parseInt(s: String): Option<Int> =
+            Try { s.toInt() }.fold(ifFailure = { None }, ifSuccess = { v -> Some(v) })
+
+    fun validateLogin(cred: Credential): Either<SecurityError, Unit>
+
+    fun userInfo(user: User): DeferredK<Profile>
+}
+```
+
+As we can see, every function above only takes as parameter the argument to perform its operation on.
+
+The next objects are just for mocking and simulate results of a side effect performed by an external system so we can see the output of our examples.
+
+```kotlin:ank:silent
+object ValidEffects : OurSideEffectsFunctions {
+
+    override fun validateLogin(cred: Credential): Either<SecurityError, Unit> {
+        return Either.right(Unit)
+    }
+
+    override fun userInfo(user: User): DeferredK<Profile> {
+        return DeferredK.async { Profile(id = user.id) }
+    }
+
+    fun savingProfiles(): DeferredK<Unit>  = GlobalScope.async(Dispatchers.Default, CoroutineStart.LAZY){Unit}.k()
+
+}
+```
+
+We just defined results for our functions when everything went "right", the next object is also just for simulating the result when something goes "wrong".
+
+```kotlin:ank:silent
+object ErrorEffects : OurSideEffectsFunctions {
+
+    override fun validateLogin(cred: Credential): Either<SecurityError, Unit> {
+        return Either.left(SecurityError.RuntimeSecurityError("Invalid credentials"))
+    }
+
+    override fun userInfo(user: User): DeferredK<Profile> {
+        return DeferredK.failed(Throwable("Error retrieving profile"))
+    }
+}
+```
+
+If you want to run these examples you can run them inside a main body.
+
+```kotlin:ank:silent
+fun main(args: Array<String>) {
+
+}
+```
+
+If we need to extract the information for a List of Users we can create a function that composes with the function defined previously `userInfo`.
+
+```kotlin:ank
+fun profilesFor(users: List<User>): List<DeferredK<Profile>> = users.map { u -> ValidEffects.userInfo(u) }
+```
+
+Notice how we are returning a List of deferred computations. It would be nice for the ones using this function if we could aggregate the results and return the List of Profile under a single DeferredK, something like `DeferredK<List<Profile>>`
+
+To be able to do this transformation we have the `Traverse` type class.
+
+
+`Traverse` is defined with the following signature:
+
+interface Traverse<F> : Functor<F>, Foldable<F> {
```

Comment:
I can see why you're giving details about how it's implemented here, so you can explain how it should be called. 

I usually don't like to add details about how things are working under the hood, since the user should just be interested on **how to call it** and **what's the role on the library** for the function / feature. But you're basically trying to explain how to call it here.

Could we link to the corresponding higher kinds explanation on arrow docs from here? If the reader is not really aware of what's a Kind yet, he's gonna suffer a bit. He should have an easy way to go there and learn about that before stepping into this.

Reply:
I am still not sure about how to rework this, but following Raul comments below, my intention is reworking the document, and provide first the explanation of the definition briefly and then adding the examples. Otherwise, what are the higher kind explanation you are referring to, @JorgeCastilloPrz ?